test/metrics_compare_utils_test.py

```diff
@@ -0,0 +1,276 @@
+        config={'base_tolerance': 2.0})
+
+    # Since the tolerance is 2.0, the small differences in values are not
+    # big enough to trigger lines in the difference report.
+    self.assertEqual(metrics_difference_report, '')
+
+
+  def test_compare_metrics_reports_value_difference_tolerance_strict(self):
+    data_points = mcu.get_data_points_from_metrics_reports(
+        [_REPORT_3, _REPORT_3, _REPORT_3_SLIGHTLY_DIFFERENT_VALUES])
+    metrics_difference_report = mcu.compare_metrics(
+        data_points, _REPORT_3_SLIGHTLY_DIFFERENT_VALUES,
+        config={'base_tolerance': 0.0})
+
+    # Since the tolerance is 0.0, even a tiny difference leads to a line in
+    # the metrics_difference_report.
+    expected_report = 'InboundData__Accumulator_mb is outside the expected range using tolerance: 0.0. Lower limit: 68083.33333333333  Upper limit: 68083.33333333333  Actual Value: 74750.0\nInboundData__TotalSamples is outside the expected range using tolerance: 0.0. Lower limit: 72144.0  Upper limit: 72144.0  Actual Value: 70000.0\nUniqueCounter__Value is outside the expected range using tolerance: 0.0. Lower limit: 9333.0  Upper limit: 9333.0  Actual Value: 9999\n'
+    self.assertEqual(metrics_difference_report, expected_report)
+
+
+  def test_compare_metrics_reports_value_difference_tolerance_custom(self):
+    data_points = mcu.get_data_points_from_metrics_reports(
+        [_REPORT_3, _REPORT_3, _REPORT_3_SLIGHTLY_DIFFERENT_VALUES])
+    metrics_difference_report = mcu.compare_metrics(
+        data_points, _REPORT_3_SLIGHTLY_DIFFERENT_VALUES,
+        config={
+            'base_tolerance': 0.0,
+            'InboundData__Accumulator_mb_tolerance': 2.0,
+            'UniqueCounter__Value_tolerance': 1.5,
+        })
+
+    # 2 of 3 differing values have custom tolerances and therefore should pass.
+    # The third uses the default tolerance of 0.0, so it will generate a line.
+    expected_report = 'InboundData__TotalSamples is outside the expected range using tolerance: 0.0. Lower limit: 72144.0  Upper limit: 72144.0  Actual Value: 70000.0\n'
+    self.assertEqual(metrics_difference_report, expected_report)
+ 
+
+  def test_compare_metrics_reports_new_counters(self):
+    data_points = mcu.get_data_points_from_metrics_reports(
+        [_REPORT_3, _REPORT_3, _REPORT_3_SLIGHTLY_DIFFERENT_VALUES])
+    metrics_difference_report = mcu.compare_metrics(
+        data_points, _REPORT_3_WITH_NEW_COUNTERS,
+        config={'base_tolerance': 2.0})
+
+    # Since the tolerance is 2.0, the small differences in values are not
+    # big enough to trigger lines in the difference report.
+    expected_report = 'Found new aten counter: aten::_local_scalar_dense__Value: 73216\n'
+    self.assertEqual(metrics_difference_report, expected_report)
+
+  def test_parse_real_metrics(self):
+    print("Testing against TPU. If this hangs, check that $XRT_TPU_CONFIG is set")
+    x = torch.rand(3, 5, device=xm.xla_device())
```

Comment:
Please do more ops here, to make sure we capture more metrics (and lower the "surprise rate" 😄 ).

Reply:
Added a few more and verified that the list of metrics became longer